bcgov/biohubbc
# SIMSBIOHUB-61: See Error Message When Submitting Artifacts Fails

# Overview

When a request to submit data to Biohub fails, the user should see an appropriate error dialog.

## Links to Jira tickets

 - https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-61

## Description of relevant changes

 - Removed duplicate code by creating the `LoadingButton` component as a stand-in for the analogous MUI Lab version
 - Refactored the Project and Survey submission components
 - Added an error dialog when the request to submit to Biohub fails

## Testing Procedure
0. Sign into SIMS in an environment that procludes submitting to Biohub (or set `BACKBONE_INTAKE_ENABLED` to false).
1. Create or navigate to a project that has unsubmitted documents.
2. Submit at least one or more documents to Biohub. You should:

 - [x] See an error dialog.

3. Create or navigate to a survey that has unsubmitted artifacts.
4. Submit at least one or more artifacts, summary results or observations data to Biohub. You should:

 - [x] See an error dialog.
